src/state/shared/assets.ts

```diff
@@ -0,0 +1,36 @@
+import { makeAutoObservable, runInAction, when } from 'mobx';
+import { ViewService } from '@penumbra-zone/protobuf';
+import { Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { getDenomMetadata } from '@penumbra-zone/getters/assets-response';
+import { penumbra } from '@/utils/penumbra';
+import { connectionStore } from '@/state/connection';
+
+class AssetsState {
+  /** If true, ignore all other state values */
+  loading = false;
+  /** The error is set in case of a request failure */
+  error?: string;
+  assets: Metadata[] = [];
+
+  constructor() {
+    makeAutoObservable(this);
+
+    when(
+      () => connectionStore.connected,
+      () => void this.setup(),
```

Comment:
question: I did this pattern on the connection status/block sync indicator PR. As it is on every page, it can always run. However, if we continue this pattern with all of our stores, won't that make a connection action request possibly too much when the view they are looking at doesn't require it?

Aka, should we instead have a useEffect() hook within react that initiates the setup? So that the lifecycle of the component governs the data loading?

Reply:
The way I see it, the stores are initiated right where they are imported for the first time. This way, if defined and used correctly (we can define the rules about it), the stores will set up when needed or triggered by the user